### PR TITLE
Add possibility to set the number of transfer and the transfer size

### PR DESCRIPTION
**Pull request by pbodilis**
Right now, current libuvc create 100 buffers of 16 * 1024  * 1024 bytes, plus as many transfers. Unsurprisingly, this tends to fail quite often on non-root android. This PR makes the default number of transfers and buffers as well as the buffer size configurable by the user.

### CODE REVIEW

File: src/init.c [hunk 1]
```diff
@@ -161,3 +161,27 @@ void uvc_start_handler_thread(uvc_context_t *ctx) {
...
 
+static uvc_log_func_t *uvc_log_func = NULL;
+#define UVC_LOG_LINE_MAX_LEN 2048
+void uvc_log_set_function(uvc_log_func_t func) {
+    uvc_log_func = func;
+}
+
+#include <stdarg.h>
+#include <libgen.h>
+void uvc_log(char *filename, unsigned line, const char *function, const char *format, ...) {
```
Comment (waddlesplash) [inline src/init.c]: These changes appear unrelated; please submit in a separate PR.

File: include/libuvc/libuvc_internal.h [hunk 2]
```diff
@@ -214,9 +214,6 @@ typedef struct uvc_device_info {
   We could/should change this to allow reduce it to, say, 5 by default
   and then allow the user to change the number of buffers as required.
  */
```
Comment (waddlesplash) [inline include/libuvc/libuvc_internal.h]: This comment no longer belongs here now that the constants have been moved.

**Comment by waddlesplash** [comment]
I think reducing the default number of buffers from 100 to 30 (1 second or so) probably makes sense, also, if you want to make this change at the same time.

**Comment by openastroproject** [comment]
I think originally the number of buffers was 5.  I bumped it up to 100 some time back because I was having problems with UVC cameras with large RGB frame sizes running at 60fps if I recall correctly.

The ability to set these limits dynamically is an improvement without a doubt.  Not sure about the default values though.  I believe the failure mode isn't one that leads to the obvious conclusion that these values need to be raised.  Unfortunately I don't have access to those cameras at the moment to be able to reproduce the failure.

**Comment by artyom-beilis** [comment]
Is there up-to-date version? Since it has conflicts?

**Comment by artyom-beilis** [comment]
I have problems using YUV2 streams, even at linux full frame 1920x1080 on some webcameras fails while at Android situation is far worse.

**Comment by jeanpiere-demir-oxb** [comment]
> Is there up-to-date version? Since it has conflicts?

I would be really interested with that answer too!

**Comment by pbodilis** [comment]
will open a new PR for logs
I'll check how the changes to number & size of buffer work
